tools/deviceGen/input/device_features.json

```diff
@@ -0,0 +1,2666 @@
+            "name": "fillModeNonSolid",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] good device support"
+        },
+        {
+            "type": "bool",
+            "name": "depthBounds",
+            "value": false,
+            "comment": "this is kinda like a weird clip-plane that doesn't count towards clip plane count"
+        },
+        {
+            "type": "bool",
+            "name": "wideLines",
+            "value": false,
+            "comment": "good device support, but a somewhat useless feature (constant screenspace width with limits on width)"
+        },
+        {
+            "type": "bool",
+            "name": "largePoints",
+            "value": false,
+            "comment": "good device support, but a somewhat useless feature (axis aligned screenspace boxes with limits on size)"
+        },
+        {
+            "type": "bool",
+            "name": "alphaToOne",
+            "value": true,
+            "comment": "Some AMD don't support"
+        },
+        {
+            "type": "bool",
+            "name": "multiViewport",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] good device support"
+        },
+        {
+            "type": "bool",
+            "name": "samplerAnisotropy",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] Roadmap 2022 requires support for these, device support is ubiquitous"
+        },
+        {
+            "groupComment": [
+                "[DO NOT EXPOSE] these 3 don't make a difference, just shortcut from Querying support from PhysicalDevice"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "textureCompressionETC2"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "textureCompressionASTC_LDR"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "textureCompressionBC"
+        },
+        {
+            "type": "bool",
+            "name": "occlusionQueryPrecise",
+            "value": true,
+            "expose": false,
+            "comment": "[REQUIRE] ROADMAP 2022 and good device support"
+        },
+        {
+            "type": "bool",
+            "name": "pipelineStatisticsQuery",
+            "value": false
+        },
+        {
+            "groupComment": [
+                "[EXPOSE AS LIMIT] Enabled by Default, Moved to Limits"
+            ]
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "vertexPipelineStoresAndAtomics"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "fragmentStoresAndAtomics"
+        },
+        {
+            "declare": true,
+            "type": "bool",
+            "name": "shaderTessellationAndGeometryPointSize"
```

Comment:
ok, either:
- we change `expose` into an enum of `DEFAULT/EXPOSE, HIDE, AS_LIMIT`, or
- you set the `expose=false` and then you need to manually re-add these to limits.json

Reply:
either will work, because that also makes sure that the members are compared in their respective feature/limit intersection/union methods

Reply:
however the first option would likely improve the placement of the members in the HLSL structs (better groupings)

Reply:
Done